src/autopas/tuning/utils/Smoothing.cpp

```diff
@@ -9,61 +9,40 @@
 #include "autopas/tuning/searchSpace/Evidence.h"
 #include "autopas/utils/Math.h"
 
-/**
- * Calculates the weights for the k-neighbors of the last point in the vector using the tri-cube function.
- * If the sum of weights is <= 0 all neighbors are at the same x coordinate as the point and smoothing will not change
- * the point. This case is indicated by the bool in the return tuple.
- *
- * @param points Sorted collection of observations.
- * @param pointsPerEstimation Number of neighbors to consider for smoothing.
- * @param maxDistFromIntervalEdge Largest distance between the point that shall be fitted and its neighbors.
- * @return Tuple of a vector containing the weights for the neighbors
- * and a bool indicating whether fitting is unnecessary.
- */
-std::tuple<std::vector<double>, bool> calculateWeightsSimple(const std::vector<autopas::Evidence> &points,
-                                                             size_t pointsPerEstimation,
-                                                             size_t maxDistFromIntervalEdge) {
-  // since we will only smooth the last point there is no outer loop and indexToFit shall be fixed
-  const size_t indexToFit = points.size() - 1;
-  const size_t firstIndex = indexToFit - pointsPerEstimation + 1;
+std::tuple<std::vector<double>, bool> autopas::smoothing::calculateWeightsSimple(
+    const std::vector<autopas::Evidence> &points, size_t pointsPerEstimation, size_t maxDistFromIntervalEdge) {
+  // since we will only smooth the last point there is no outer loop of LOESS and indexToFit shall be fixed
+  const auto indexToFit = points.size() - 1;
+  const auto firstIndex = indexToFit - pointsPerEstimation + 1;
   // initialize all weights with 0
   std::vector<double> weights(pointsPerEstimation);
 
-  const auto &xi = points[indexToFit].iteration;
-
-  // Define thresholds for shortcuts: If residuals are beyond these values, they
-  // are assumed to be 0, respectively 1.
-  auto maxDistFromIntervalEdgeHigh = maxDistFromIntervalEdge * .999;
-  auto maxDistFromIntervalEdgeLow = maxDistFromIntervalEdge * .001;
+  // get iteration number of the point being smoothed
+  const auto &xi = static_cast<long>(points[indexToFit].iteration);
 
   double sumOfWeights = 0.0;
 
-  // compute all weighs that can be assumed to be non-zero
-  for (size_t j = 0; j < weights.size(); ++j) {
-    auto xj = points[j + firstIndex].iteration;
+  // compute all weights that can be assumed to be non-zero
+  for (size_t j = 0; j < pointsPerEstimation; ++j) {
+    // set weight to zero
+    weights[j] = 0.;
+    const auto xj = static_cast<long>(points[j + firstIndex].iteration);
     // residual = |xi - xj|
-    const size_t residual = xj > xi ? xj - xi : xi - xj;
-
-    // if x are too far apart ...
-    if (residual <= maxDistFromIntervalEdgeHigh) {
-      if (residual > maxDistFromIntervalEdgeLow) {
```

Comment:
can you explain why you removed / replaced this? (incl the else)

Reply:
`residual > maxDistFromIntervalEdgeLow` holds always unless `residual==0`. In the old code, the `residual==0` case is handled by the `else` clause (see original lines 53-55). This sets the weight to `1.0`. Except the formula provided in the `if` clause will result in `1.0` as well if evaluated with `residual=0`. Therefore, there is no need to handle this case separately.

Except, of course, for performance reasons. But I doubt the `pow`s here are causing any noticeable performance cost so I would rather simplify the code.